src/mlpack/core/optimizers/sgd/update_policies/gradient_clipping.hpp

```diff
@@ -0,0 +1,96 @@
+/**
+ * @file gradient_clipping.hpp
+ * @author Konstantin Sidorov
+ *
+ * Gradient clipping update wrapper.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_SGD_GRADIENT_CLIPPING_HPP
+#define MLPACK_CORE_OPTIMIZERS_SGD_GRADIENT_CLIPPING_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace optimization {
+
+/**
+ * Interface for wrapping around update policies (e.g., VanillaUpdate)
+ * and feeding a clipped gradient to them instead of the normal one.
+ * (Clipping here is implemented as
+ * \f$ g_{\text{clipped}} = \max(g_{\text{min}}, \min(g_{\text{min}}, g))) \f$.)
+ * 
+ * @tparam UpdatePolicy A type of UpdatePolicy that sould be wrapped around.
+ */
+template<typename UpdatePolicy>
+class GradientClipping
+{
+ public:
+  /**
+   * Constructor for creating a GradientClipping instance.
+   * 
+   * @param minGradient Minimum possible value of gradient element.
+   * @param maxGradient Maximum possible value of gradient element.
+   * @param updatePolicy An instance of the UpdatePolicy
+   *                     used for actual optimization.
+   */
+  GradientClipping(const double minGradient,
+                   const double maxGradient,
+                   UpdatePolicy updatePolicy) :
```

Comment:
Likewise, it looks reasonable to add `MinGradient` and `MaxGradient` methods (done in the last commit)